cli/cli.py

```diff
@@ -0,0 +1,297 @@
+        self.users.remove(account)
+        self.users.append(account)
+        # save the new auth state when done
+        self.save(self.users)
+
+    def authenticate(self, username, password):
+        account, err = self.get(username)
+
+        if account is None:
+            return 'Error', err
+
+        if account['password'] == self.sha_password(password):
+            return True, 'Authenticated'
+        else:
+            return False, 'Incorrect password.'
+
+
+class Config(object):
+    def __init__(self):
+        self.config = self.load()
+
+    @classmethod
+    def load(cls):
+        try:
+            config_path = os.getcwd() + '/' + CONF_FILE
+            # make sure the file is not empty
+            if os.stat(config_path).st_size != 0:
+                conf = configparser.ConfigParser()
+                conf.read(config_path)
+                log.debug('Config file loaded: ' + config_path)
+
+                return conf
+        except (FileNotFoundError, IOError) as e:
+            log.info('Unable to open file.')
+            log.info(e)
+            exit()
+
+    def get(self, section, key):
+        return self.config.get(section, key)
+
+    def set(self, section, key, value):
+        self.config.set(section, key, value)
+
+    def show(self):
+        for section in self.config.sections():
+            for option in self.config.options(section):
+                print(option, '=', self.config.get(section, option))
+
+    def save(self):
+        with open(CONF_FILE, 'w') as f:
+            self.config.write(f)
+
+
+class Certificate(object):
+
+    def __init__(self):
+        self.auth_file = Config().get(APP_NAME, 'lxdui.auth.conf')
+        self.cert_file = Config().get(APP_NAME, 'lxdui.ssl.cert')
+        self.key, self.cert = self.create()
+
+    # Create a self signed certificate
+    @staticmethod
+    def create():
+        # create a key pair
+        k = crypto.PKey()
+        k.generate_key(crypto.TYPE_RSA, 1024)
+
+        # create a self-signed cert
+        cert = crypto.X509()
+        cert.get_subject().C = "US"
+        cert.get_subject().ST = "Texas"
+        cert.get_subject().L = "Dallas"
+        cert.get_subject().O = "AdaptiveScale, Inc."
+        cert.get_subject().OU = "OU=AdaptiveScale, DN=com"
+        cert.get_subject().CN = gethostname()
+        cert.set_serial_number(1000)
+        cert.gmtime_adj_notBefore(0)
+        cert.gmtime_adj_notAfter(10 * 365 * 24 * 60 * 60)
+        cert.set_issuer(cert.get_subject())
+        cert.set_pubkey(k)
+        cert.sign(k, 'sha1')
+
+        key = crypto.dump_privatekey(crypto.FILETYPE_PEM, k)
+        cert = crypto.dump_certificate(crypto.FILETYPE_PEM, cert)
+
+        return key, cert
+
+    @staticmethod
+    def save(file, data):
+        try:
+            with open(file, 'wb') as f:
+                f.write(data)
+        except (FileNotFoundError, IOError) as e:
+            log.info('Unable to open file.')
+            log.info(e)
```

Comment:
We can split User, Config, Certificate to separate files and include them in the models. This will allow use to reuse the models from the API also.